ext/pdo_mysql/tests/gh13384.phpt

```diff
@@ -0,0 +1,37 @@
+--TEST--
+GH-13384 Fixed GH-13167 Fixed the behavior when an inappropriate value was passed to `bindValue`.
+--EXTENSIONS--
+pdo_mysql
+--SKIPIF--
+<?php
+require_once __DIR__ . '/inc/mysql_pdo_test.inc';
+MySQLPDOTest::skip();
+?>
+--FILE--
+<?php
+require_once __DIR__ . '/inc/mysql_pdo_test.inc';
+$db = MySQLPDOTest::factory();
+$db->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
+$db->setAttribute(PDO::ATTR_EMULATE_PREPARES, false);
+
+$db->exec('CREATE TABLE test_gh13384 (mode TINYINT)');
+
+$stmt = $db->prepare('INSERT INTO test_gh13384 (mode) VALUES (?)');
+$stmt->bindValue(1, new DateTime(), PDO::PARAM_INT);
```

Comment:
What happens if it's `bindParam` instead? Does the type of the variable get cast too?